src/unix/mediactrl_gstplayer.cpp

```diff
@@ -337,6 +375,18 @@ bool wxGStreamerMediaBackend::CreateControl(wxControl* ctrl, wxWindow* parent,
     m_video_renderer = gst_player_video_overlay_video_renderer_new(window_handle);
     m_player = gst_player_new(m_video_renderer, gst_player_g_main_context_signal_dispatcher_new(NULL));
 
+    wxDisplayInfo info = wxGetDisplayInfo();
+    if (info.type == wxDisplayWayland) {
+        // wayland needs a specific handler to pass display to gstreamer
+        GstBus *bus = gst_pipeline_get_bus(GST_PIPELINE(gst_player_get_pipeline(m_player)));
+        gst_bus_add_signal_watch(bus);
+        gst_bus_set_sync_handler(bus, bus_sync_handler, this, NULL);
+        gst_object_unref(bus);
+
+        // having display set causes crash when gstreamer tries to probe xvimagesink
+        unsetenv("DISPLAY");
```

Comment:
I feel pretty uncomfortable about this because it could easily break something else relying on `DISPLAY` being set (it does seem unlikely that it's going to be used with Wayland, but unlikely doesn't mean impossible...). Could we at least save the old env var value and restore it after doing whatever we do that results in the crash?

Reply:
Right - if someone relies on the wxwidgets app to start some X11 native window on a button click or something, it'll be pretty bad.

I'll try to figure if we can tell gstreamer to not try xvimagesink somehow, and if I can't we can restore it in the realize callback I think.
Still quite bad though (there's possibly a long time between both isn't there? this would be done at window creation time, but realize would be done when playback starts afaiu...); might be possible to unset DISPLAY just before playback starts, I'll have a new look...

Reply:
Thanks for insisting on it, I wasn't too happy about it either.
I found a way to tell gstreamer not to load a plugin and changed this for that, so we no longer need to mess with DISPLAY.